src/opm/io/eclipse/ESmry_write_rsm.cpp

```diff
@@ -0,0 +1,198 @@
+    const std::string divider_line { std::string(total_width - 3, '-') } ;
+
+    const std::string block_header_line(const std::string& run_name, const std::string& comment) {
+        return "SUMMARY OF RUN " + run_name + " OPM FLOW VERSION 1910 " + comment;
+    }
+
+    void write_line(std::ostream& os, const std::string& line, char prefix = ' ') {
+        os << prefix << std::setw(total_width) << std::left << line << '\n';
+    }
+
+    void print_text_element(std::ostream& os, const std::string& element) {
+        os << std::setw(8) << std::left << element << std::setw(5) << "";
+    }
+
+    void print_float_element(std::ostream& os, float element) {
+        static const std::regex integer_regex { "\\.0*$" };
+
+        auto element_string = std::to_string(element);
+        element_string = std::regex_replace(element_string, integer_regex, "");
+
+        if (element_string.size() > 8) {
+            element_string = element_string.substr(0, 8);
+        }
+
+        os << std::setw(8) << std::right << element_string << std::setw(5) << "";
+    }
+
+    void write_header_columns(std::ostream& os, const std::vector<Opm::EclIO::SummaryNode>& vectors, std::function<void(std::ostream&, const Opm::EclIO::SummaryNode&)> print_element, char prefix = ' ') {
+        os << prefix;
+
+        for (const auto& vector : vectors) {
+            print_element(os, vector);
+        }
+
+        os << '\n';
+    }
+
+    void write_data_row(std::ostream& os, const std::vector<std::pair<std::vector<float>, int>>& data, std::size_t index, char prefix = ' ') {
+        os << prefix;
+
+        for (const auto& vector : data) {
+            print_float_element(os, vector.first[index] * std::pow(10.0, -vector.second));
+        }
+
+        os << '\n';
+    }
+
+    void write_scale_columns(std::ostream& os, const std::vector<std::pair<std::vector<float>, int>> data, char prefix = ' ') {
+        os << prefix;
+
+        for (const auto& vector : data) {
+            const auto scale_factor { vector.second } ;
+            if (scale_factor) {
+                print_text_element(os, "*10**" + std::to_string(scale_factor));
+            } else {
+                print_text_element(os, "");
+            }
+        }
+
+        os << '\n';
+    }
+
+}
+
+namespace Opm::EclIO {
+
+void ESmry::write_block(std::ostream& os, const std::vector<SummaryNode>& vectors) const {
+    write_line(os, version_line, '1');
+    write_line(os, divider_line);
+    write_line(os, block_header_line(inputFileName.stem(), "ANYTHING CAN GO HERE: USER, MACHINE ETC."));
+    write_line(os, divider_line);
+
+    std::vector<std::pair<std::vector<float>, int>> data;
+
+    bool has_scale_factors { false } ;
+    for (const auto& vector : vectors) {
+        int scale_factor { 0 } ;
+        const auto& vector_data { get(vector) } ;
+
+        auto max = *std::max_element(vector_data.begin(), vector_data.end());
+
+        if (max >= 10'000'000'000) {
```

Comment:
I guess this could be something like: `scale_factor = 3*std::floor( (std::log10( max ) - 4)/3 )`